rs/rosetta-api/icp_ledger/ledger/tests/tests.rs

```diff
@@ -190,6 +191,80 @@ fn test_minting_account() {
     ic_icrc1_ledger_sm_tests::test_minting_account(ledger_wasm(), encode_init_args)
 }
 
+#[test]
+fn test_anonymous_transfers() {
+    const INITIAL_BALANCE: u64 = 10_000_000;
+    const TRANSFER_AMOUNT: u64 = 1_000_000;
+    let p1 = PrincipalId::new_user_test_id(1);
+    println!("p1: {:?}", p1);
+    let anon = PrincipalId::new_anonymous();
+    println!("anon: {:?}", anon);
+    let (env, canister_id) = setup(
+        ledger_wasm(),
+        encode_init_args,
+        vec![(Account::from(p1.0), INITIAL_BALANCE)],
+    );
+
+    assert_eq!(INITIAL_BALANCE, total_supply(&env, canister_id));
+    assert_eq!(INITIAL_BALANCE, balance_of(&env, canister_id, p1.0));
+    assert_eq!(0u64, balance_of(&env, canister_id, anon.0));
+
+    // Transfer to the account of the anonymous principal succeeds
+    transfer(&env, canister_id, p1.0, anon.0, TRANSFER_AMOUNT).expect("transfer failed");
```

Comment:
Should this not also fail?
Otherwise it's essentially an unregistered brun operation.

Reply:
I suppose that's TBD - here I'm only adding tests to verify the current behavior.